src/penalty_parameter.jl

```diff
@@ -0,0 +1,48 @@
+"""
+Abstract type for ProgressiveHedging penalty parameter.
+
+Concrete sybtypes determine how this penalty is used within the PH algorithm.
+"""
+abstract type AbstractPenaltyParameter end
+
+## Scalar Penalty Implementation ## 
+
+struct ScalarPenaltyParameter{T<:Real} <: AbstractPenaltyParameter
+    value::T
+end
+convert(::Type{T}, r::ScalarPenaltyParameter{S}) where {T<:Real, S<:Real} = T(r.value)
+
+## Proportional Penalty Implementation ##
+struct ProportionalPenaltyParameter{T<:Real} <: AbstractPenaltyParameter
+    value::T
+end
+convert(::Type{T}, r::ProportionalPenaltyParameter{S}) where {T<:Real, S<:Real} = T(r.value)
+
+# NOTE: temporary fcns for coefficients of expressions
```

Comment:
The functions below here are actually subproblem dependent (they would be different for another concrete subproblem type--i.e. something that doesn't use JuMP as its algebraic modeling language) so they should actually stay with the `JuMPSubproblem` code in subproblem.jl.  I should have been more specific before. Sorry.